ui/src/app/services/patch-db/patch-db.service.ts

```diff
@@ -128,22 +150,37 @@ export class PatchDbService {
     const argsString = '/' + args.join('/')
     console.log('patchDB: WATCHING ', argsString)
     return this.patchDb.store.watch$(...(args as []))
-    .pipe(
-      tap(data => console.log('patchDB: NEW VALUE', argsString, data)),
-      catchError(e => {
-        console.error('patchDB: WATCH ERROR', e)
-        return of(e.message)
-      }),
-      finalize(() => console.log('patchDB: UNSUBSCRIBING', argsString)),
-    )
-  }
-
-  private async initSource (source: Source<DataModel>): Promise<void> {
-    this.stop()
-    const cache = await this.bootstrapper.init()
-    console.log('CACHE', cache)
-    this.patchDb = new PatchDB([source, this.http], this.http, cache)
-    this.data = this.patchDb.store.cache.data
-    console.log('DATA', this.data)
+      .pipe(
+        tap(data => console.log('patchDB: NEW VALUE', argsString, data)),
+        catchError(e => {
+          console.error('patchDB: WATCH ERROR', e)
+          return of(e.message)
+        }),
+        finalize(() => console.log('patchDB: UNSUBSCRIBING', argsString)),
+      )
   }
 }
+
+
+function switchMapAfter<A, B> (projection: (a: A) => Observable<B>) {
+  return (observable: Observable<A>) =>
+    new Observable<B>((subscriber) => {
+      let lastLive: null | Subscription = null
+      const subscription = observable.subscribe({
+        next (value) {
+          console.log('switchMapAfter JCWM new events')
```

Comment:
Remove comment

Reply:
Maybe it's time for a proper level/sink-based logging library for the FE.